### DISCUSSION TITLE
How to convert battery voltage or battery current into level of percentage?

### DISCUSSION THREAD

**Discussion by artpoz** [pybricks/support#1675] [Q&A] [answered]
Hi.

How to convert battery voltage or battery current into level of percentage? (from 0 to 100)

Best regards
artpoz

**Comment by Debenben** [comment]
You could record a discharge curve for your battery at a given temperature and load and then map voltage values accordingly. What hub are and battery are you using?

**Reply by artpoz** [reply]
I've got Inventor hub and rechargable battery in #51515 set. Fully charged battery has about 8000 mV.

**Comment by dlech** [comment, marked as answer]
There are some interesting voltages listed here: https://github.com/pybricks/pybricks-micropython/blob/d16a98f1e74fefe030ce6cd0981c92339662ce94/lib/pbio/sys/battery.c#L24-L28

The voltage doesn't have a linear relation with the amount of energy remaining, so if you want something accurate, you would need to make a discharge curve like @Debenben said. Or if you just want to know close to full or close to empty, you could divide it into 5 levels (0%, 20%, 40%, 80%, 100%) and use the voltages in the link as a guide.

**Comment by artpoz** [comment]
I made program that shows battery level.

```python
from pybricks.hubs import InventorHub
from pybricks.pupdevices import Motor, ColorSensor, UltrasonicSensor
from pybricks.parameters import Button, Color, Direction, Port, Side, Stop
from pybricks.robotics import DriveBase
from pybricks.tools import wait, StopWatch

hub = InventorHub()
current_voltage = hub.battery.voltage()

def show_battery_level():
    
    hub.display.off()    
    led_on_crit_full = [6000, 6400, 6800, 7200, 8300]

    for led_y in range (5):  
        for led_x in range (5):
            if current_voltage >= led_on_crit_full[led_y]:
                hub.display.pixel(led_x,led_y,100)                             
       
    wait(3000)
show_battery_level()
```
